scripts/src/java/org/oppia/android/scripts/coverage/RunCoverageForTestTarget.kt

```
@@ -65,7 +66,7 @@ class RunCoverageForTestTarget(
   /**
    * Analyzes target file for coverage, generates chosen reports accordingly.
    */
-  fun runCoverage(): List<String>? {
+  fun runCoverage(): CoverageReport? {
```

Here & elsewhere: we probably should throw an exception rather than return null if we can produce a report (especially since that should only happen in exceptional cases).

---

Refactored it now as to throw RuntimeException
```
fun runWithCoverageAsync(
    bazelTestTarget: String
  ): Deferred<CoverageReport> {
    return CoroutineScope(scriptBgDispatcher).async {
      val coverageResult = retrieveCoverageResult(bazelTestTarget)
        ?: throw RuntimeException("Failed to retrieve coverage result for $bazelTestTarget")
}
```

---

Would this also apply to the CoverageRunners's retrieveCoverageResult that retrieves the result from bazelClient?
```
  private fun retrieveCoverageResult(
    bazelTestTarget: String
  ): List<String>? {
    return bazelClient.runCoverageForTestTarget(bazelTestTarget)
  }
```

---

Now it uses - `error("Failed to retrieve coverage result for $bazelTestTarget")`

---

This file is nolonger in this PR. Did you move it to a different one?

---

Actually, the **file was removed** to keep the scripts clean and non-redundant.

Original implementation involved introducing 3 scripts, but then felt the 3rd one redundant, so it was then decided to be removed as per https://github.com/oppia/oppia-android/pull/5433#discussion_r1648659133.

The above mentioned `error` change was updated in other relevant references in the script: https://github.com/oppia/oppia-android/pull/5439/files#diff-e1015bdd1aafa12e3e95de3b408525733c702992881ece1d2399b57ded6a44fdR42